packages/docs-reanimated/docs/reanimated-babel-plugin/about.md

```diff
@@ -0,0 +1,217 @@
+
+- `useAnimatedStyle`
+- `useAnimatedProps`
+- `runOnUI`
+
+etc.
+
+Thanks to that, you don't need to add the `'worklet'` directive to your code when you're not exploring any advanced use cases.
+
+### Referencing worklets
+
+You can define worklets before they are used, and the plugin will autoworkletize them too:
+
+```ts
+function foo() {
+  // You don't need to mark it as a worklet.
+  return { width: 100 };
+}
+
+const style = useAnimatedStyle(foo); // You don't need to define an inline function here.
+```
+
+### Objects aggregating worklets
+
+In some APIs, like `useAnimatedScrollHandler` you can pass an object that contains worklets instead of a function:
+
+```ts
+const handlerObject = {
+  // You don't need to mark these methods as worklets.
+  onBeginDrag() {
+    console.log('Dragging...');
+  },
+  onScroll() {
+    console.log('Scrolling...');
+  },
+};
+
+const handler = useAnimatedScrollHandler(handlerObject);
+```
+
+### Workletizing whole files (experimental)
+
+You can mark a file as a workletizable file by adding the `'worklet'` directive to the top of the file. This will make all _top-level_ entities workletized. This can come in handy for files that contain multiple worklets.
+
+```ts
+'worklet';
+
+function foo() {
+  // This function will be autoworkletized.
+  return { width: 100 };
+}
+
+function bar() {
+  // This function will be autoworkletized.
+  function foobar() {
+    // This function won't since it's not defined in top-level scope.
+    console.log('Hello from worklet');
+  }
+  return { width: 100 };
+}
+```
+
+## Limits of autoworkletization
+
+The plugin cannot infer whether a function is autoworkletizable or not in some contexts.
+
+### Imports
+
+When importing a function from another file or a module and using it as a worklet, you must manually add the `'worklet'` directive to the function:
+
+```ts
+// foo.js
+export function foo() {
+  'worklet'; // Won't work without it.
+  return {
+    width: 100,
+  };
+}
+
+// bar.ts
+import { foo } from './foo';
+const style = useAnimatedStyle(foo);
+```
+
+### Custom hooks
+
+Currently Reanimated hasn't exposed APIs that would allow you to register your custom hooks for callback workletization. This however, might change in the future.
+
+### Expressions
+
+When the worklet is a result of an expression, it won't get autoworkletized hence you need to add the `'worklet'` directive to it manually:
+
+```ts
+const foo = someCondition
+  ? () => {
+      'worklet'; // Won't work without it.
+      return { width: 100 };
+    }
+  : () => {
+      'worklet'; // Won't work without it.
+      return { width: 200 };
+    };
+
+const style = useAnimatedStyle(foo);
+```
+
+In such cases we recommend you to handle the conditional logic in the worklet itself or to refactor your code to avoid the need for conditional worklets.
```

Comment:
just loose suggestion, something felt off here with 'to refactor your code to ...'

```suggestion
In such cases we recommend either handling the conditional logic in the worklet itself or refactoring your code to eliminate the need for conditional worklets.
```

Reply:
I wanted to use active voice as our guru @kacperkapusciak has taught us.

Reply:
i'm still learning this mystic knowledge, sorry :c